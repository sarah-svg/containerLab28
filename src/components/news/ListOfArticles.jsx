import React from 'react';
import PropTypes from 'prop-types';
import OneArticle from './OneArticle';

const ListOfArticles = ({ articles }) => {
  const articleInfo = articles.map(article => (
    <li key={article.publishedAt}>
      <OneArticle {...article} />
    </li>
  ));

  return (
    <ul>
      {articleInfo}
    </ul>
  );
};

ListOfArticles.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    publishedAt: PropTypes.string.isRequired
  })).isRequired
};

export default ListOfArticles;
