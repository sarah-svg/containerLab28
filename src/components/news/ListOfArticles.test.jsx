import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ListOfArticles from './ListOfArticles';


describe('ArticlesList component', () => {
  afterEach(() => cleanup());
  it('renders by mapping through and listing the articles', () => {
    const { asFragment } = render(<ListOfArticles articles={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
