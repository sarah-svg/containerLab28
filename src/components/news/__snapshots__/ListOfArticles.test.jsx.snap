// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList component renders  1`] = `
<DocumentFragment>
  <ul />
</DocumentFragment>
`;

exports[`ArticlesList component renders ArticlesList 1`] = `
<DocumentFragment>
  <ul />
</DocumentFragment>
`;

exports[`ArticlesList component renders by mapping through and listing the articles 1`] = `
<DocumentFragment>
  <ul />
</DocumentFragment>
`;
